ssh.go

```diff
@@ -70,7 +70,7 @@ type LocalUnixForwardingCallback func(ctx Context, socketPath string) bool
 
 // ReverseUnixForwardingCallback is a hook for allowing reverse unix forwarding
 // (streamlocal-forward@openssh.com).
-type ReverseUnixForwardingCallback func(ctx Context, socketPath string) bool
+type ReverseUnixForwardingCallback func(ctx Context, socketPath string) func(socketPath string) (net.Listener, error)
```

Comment:
I don't understand the need for returning a function here, it seems better to just make our DefaultSocketListener function take a context that it doesn't use to avoid the wrap